### ISSUE TITLE
Incorrect interpretation of .gitignore

### ISSUE THREAD

**Issue by adriangb** [psf/black#2084]
I like to structure my `.gitignore`'s so that I exclude everything by default and then manually include my source code:

```
/*
!/src
```

Git interprets this just fine, `src` gets checked in and not ignored.

However, black seems to think that `src` is excluded. I put a `test.py` in `src/test.py` to check:

```
$ black --verbose .
src/test.py ignored: matches the .gitignore file content
No Python files are present to be formatted. Nothing to do 😴
```

This is with `black --version` giving `20.8b1`.

**Comment by ambv** [comment]
Can you check what version of pathspec you have in your environment? We're not parsing .gitignore by hand, we're using pathspec to do it for us.

**Comment by adriangb** [comment]
Hmm I tested this again, I can no longer reproduce. It's very well possible that this was due to `pathspec` being set by some other dependency. And now I don't even remember what environment this was happening in.

In any case, with a clean install of `20.8b1` in a virtual env, I can no longer reproduce this. I'm going to close since it seems like it is not a black problem, I'll re-open if I ever run into this again.
